### PR TITLE
fix(spans): Support new attr format

### PR DESCRIPTION
**Pull request by wmak**
- Attrs before sep 23rd have the format `attr_str[something]`, after sep 23rd if they're sentry attrs they become `attr_str[sentry.something]`
- This means until December 22nd we'll have to query both
- Can't just do `has(attr_str, "sentry.something")` to know what to query since `attr_str` isn't a real column in Clickhouse, instead its something like `attr_str_0`
- Instead settling on doing a pseudo coalesce by checking the column isn't empty if we query for it

### CODE REVIEW

File: src/sentry/search/events/builder/base.py [hunk 1]
```diff
@@ -1309,7 +1309,7 @@ def default_filter_converter(
         if search_filter.operator in ("=", "!=") and search_filter.value.value == "":
             if is_tag or is_attr or is_context or name in self.config.non_nullable_keys:
                 return Condition(lhs, Op(search_filter.operator), value)
-            else:
+            elif isinstance(lhs, Column):
```
Comment (wmak) [inline src/sentry/search/events/builder/base.py:1312]: This should be column since it won't really work for Functions anyways

File: src/sentry/search/events/builder/spans_indexed.py [hunk 2]
```diff
@@ -89,25 +89,67 @@ def resolve_field(self, raw_field: str, alias: bool = False) -> Column:
...
-            return super().resolve_field(raw_field, alias)
+            # Temporary until at least after 22 Dec 2024 when old data rotates out, otherwise we should just call super
+            # here and return default_field without any extra work
+            default_field = super().resolve_field(raw_field, alias)
+            if (
+                isinstance(default_field, Column)
+                and default_field.subscriptable == "attr_str"
+                or isinstance(default_field, AliasedExpression)
+                and default_field.exp.subscriptable == "attr_str"
+            ):
```
Comment (wmak) [inline src/sentry/search/events/builder/spans_indexed.py:100]: Thanks I hate it. But its only temporary i guess?

**Comment by colin-sentry** [comment]
mapContains works in EAP dataset, it should be semantically very similar to `has` for this

File: tests/snuba/api/endpoints/test_organization_events_span_indexed.py [hunk 3]
```diff
@@ -525,7 +525,9 @@ def test_query_for_missing_tag(self):
         assert response.data["data"] == [{"foo": "", "count()": 1}]
 
 
-@pytest.mark.xfail(reason="Snuba is prefixing keys, and Sentry wasn't updated first")
+@pytest.mark.xfail(
```
Comment (colin-sentry) [inline tests/snuba/api/endpoints/test_organization_events_span_indexed.py:528]: not a huge fan of xfailing the test for the actual change you're making, let's just do this in another PR if we break something in snuba again?

**Review by colin-sentry — CHANGES_REQUESTED** [review]

File: tests/snuba/api/endpoints/test_organization_events_span_indexed.py (hunk 3 above)
Comment (colin-sentry) [reply tests/snuba/api/endpoints/test_organization_events_span_indexed.py:528]: if we *plan* on breaking that is, now that we know which tests are affected by EAP changes.

Comment (wmak) [reply tests/snuba/api/endpoints/test_organization_events_span_indexed.py:528]: I disagree, I've had to xfail tests in this file on multiple occasions already. I'm the only person working on this, and these tests don't block Snuba from merging. Which means they only need to pass as I'm working on them locally.i

Comment (wmak) [reply tests/snuba/api/endpoints/test_organization_events_span_indexed.py:528]: Unless we add these tests as something that needs to pass as part of Snuba's CI I don't see how removing the xfail helps

Comment (colin-sentry) [reply tests/snuba/api/endpoints/test_organization_events_span_indexed.py:528]: would you xfail them in a separate PR so that we can be confident CI passes for this change at least?

Comment (colin-sentry) [reply tests/snuba/api/endpoints/test_organization_events_span_indexed.py:528]: aren't there versions of these dests that run against dist tables that you can't run locally?

Comment (wmak) [reply tests/snuba/api/endpoints/test_organization_events_span_indexed.py:528]: > aren't there versions of these dests that run against dist tables that you can't run locally?

Not that I'm aware of, I've never seen a test pass locally and fail in CI

**Review by colin-sentry — APPROVED** [review]

File: src/sentry/search/events/builder/base.py (hunk 1 above)
Comment (Zylphrex) [reply src/sentry/search/events/builder/base.py:1312]: What happens if this is an alias like a field alias or alised field?
